core/chains/evm/logpoller/log_poller.go

```diff
@@ -347,6 +353,31 @@ func (lp *logPoller) run() {
 				start = lastProcessed.BlockNumber + 1
 			}
 			lp.pollAndSaveLogs(lp.ctx, start)
+		case <-backupLogPollTick:
+			// If pollPeriod is set to 1 block time, backup log poller will run once every 100 blocks
+			const backupPollerBlockDelay = 100
+
+			backupLogPollTick = time.After(utils.WithJitter(backupPollerBlockDelay * lp.pollPeriod))
+
+			latestBlock, err := lp.ec.HeadByNumber(lp.ctx, nil)
+			if err != nil {
+				lp.lggr.Warnw("backup logpoller failed to get latest block", "err", err)
+				continue
+			}
+			latestBlockNumber := latestBlock.Number
+			if firstUnfinalizedBlockNumber == 0 {
```

Comment:
100 blocks behind head may still not be the firstUnfinalized block. I think its more confusing to have the check for 
lastSafeBackfillBlock >= currentBlockNumber inside this helper method. I'd suggest just not having that helper and then in this backupPoll, just doing this:
```
backupLogPollTick = time.After(utils.WithJitter(100 * lp.pollPeriod))
latestBlock, err := lp.ec.HeadByNumber(lp.ctx, nil)
if err != nil {
	lp.lggr.Warnw("backup logpoller failed to get latest block", "err", err)
	continue
}
lastSafeBackfillBlock := latestBlock.Number - lp.finalityDepth - 1
startBlock := lp.lastBackupPollBlock + 1
if startBlock == 0 {
	startBlock = lastSafeBackfillBlock
}
if startBlock <= lastSafeBackfillBlock {
	if err := lp.backfill(ctx, startBlock, lastSafeBackfillBlock); err != nil {
		// log error and do not update lastBackupPollBlock
                continue
	}
        lp.lastBackupPollBlock = lastSafeBackfillBlock
}
```
The idea being we only ever backfill finalized blocks and never miss one (and check every 100 blocks)

Reply:
If just after startup we only go back to `startBlock = latestBlockNumber - finalityDepth`, wouldn't we miss logs much more easily?  Using latestBlockNumber - 100 is much safer than say, latestBlockNumber - 1.  We could increase the 100 if you think it's not enough, be seems dangerous to decrease it.  If the node goes down just after receiving incomplete logs from a block, then they will never get updated unless it comes back up within finalityDepth blocks.  Am I missing something?

Reply:
yeah fair enough, less protection for really short finality depths. Can keep the above code the same and just have 
lastSafeBackfillBlock := min(latestBlock.Number - lp.finalityDepth - 1, latestBlock.Number - 100)